grpc-gcp/src/main/java/com/google/cloud/grpc/GcpManagedChannel.java

```diff
@@ -982,44 +977,20 @@ public synchronized ConnectivityState getState(boolean requestConnection) {
    * channel.
    */
   protected void bind(ChannelRef channelRef, List<String> affinityKeys) {
-    synchronized (bindLock) {
-      if (affinityKeys != null && channelRef != null) {
-        for (String affinityKey : affinityKeys) {
-          affinityKeyToChannelRef.putIfAbsent(affinityKey, channelRef);
-        }
-        channelRef.affinityCountIncr(affinityKeys.size());
+    for (String affinityKey : affinityKeys) {
+      while (affinityKeyToChannelRef.putIfAbsent(affinityKey, channelRef) != null) {
+        unbind(Collections.singletonList(affinityKey));
       }
+      channelRef.affinityCountIncr();
     }
   }
 
-  /**
-   * Unbind channel with affinity key. If one ChannelRef has zero affinity key bound with it, delete
-   * all the <affinityKey, channel> maps.
-   */
+  /** Unbind channel with affinity key. */
   protected void unbind(List<String> affinityKeys) {
-    synchronized (bindLock) {
-      if (affinityKeys != null) {
```

Comment:
why remove this check. is input param guaranteed to be non-null?

Reply:
Yes, there are two places where this is called from and we already check there that it is not null.